lib/useful.ex

```diff
@@ -58,21 +82,21 @@ defmodule Useful do
 
   # avoids the error "** (Protocol.UndefinedError) protocol Enumerable
   #   not implemented for ~U[2017-08-05 16:34:08.003Z] of type DateTime"
-  defp process({key, %Date{} = date}), do: format({key, date})
-  defp process({key, %DateTime{} = datetime}), do: format({key, datetime})
+  defp process({key, %Date{} = date}), do: {key, date}
+  defp process({key, %DateTime{} = datetime}), do: {key, datetime}
 
   # process nested maps
   defp process({key, sub_map}) when is_map(sub_map) do
     for {sub_key, value} <- flatten_map(sub_map) do
-      format({"#{key}__#{sub_key}", value})
+      {"#{key}__#{sub_key}", value}
```

Comment:
@nelsonic I had to stop converting the keys to atoms to avoid crashing the BEAM on huge maps. It only converts the keys on line 49 now.